cmake/get_cpm.cmake

```diff
@@ -1,11 +1,15 @@
 set(CPM_DOWNLOAD_VERSION 1.0.0-development-version)
 
-if(CPM_SOURCE_CACHE)
-  set(CPM_DOWNLOAD_LOCATION "${CPM_SOURCE_CACHE}/cpm/CPM_${CPM_DOWNLOAD_VERSION}.cmake")
-elseif(DEFINED ENV{CPM_SOURCE_CACHE})
-  set(CPM_DOWNLOAD_LOCATION "$ENV{CPM_SOURCE_CACHE}/cpm/CPM_${CPM_DOWNLOAD_VERSION}.cmake")
-else()
-  set(CPM_DOWNLOAD_LOCATION "${CMAKE_BINARY_DIR}/cmake/CPM_${CPM_DOWNLOAD_VERSION}.cmake")
+if ( NOT CPM_DOWNLOAD_LOCATION)
```

Comment:
An unwanted side effect of keeping the value of `CPM_DOWNLOAD_LOCATION` is that users would no longer receive a warning when a upstream package uses a newer CPM version than the outer project. This is currently handled inside the newer CPM.cmake script that would be included.

Reply:
I think I can just about see the issue you refer to here but maybe not, patient with me for  a moment....

I may not be right, but for my current use-case I am using `CPM_DOWNLOAD_LOCATION` to override a projects CPM version at the top of the build. However until all the packages on which it depends are also updated to contain this check the CPM version will not be set across the entire build. However when/if this change is integrated into all packages then they would all end up using the _same_ CPM script file. 

In my view this being able to set the entire build to use a single CPM script feels preferential for developing than to supporting and testing having mixing of CPM versions across packages? It also allows testing if a dependency would have issues upgrading to a new CPM for instance with ease etc.

Reply:
> I may not be right, but for my current use-case I am using CPM_DOWNLOAD_LOCATION to override a projects CPM version at the top of the build. However until all the packages on which it depends are also updated to contain this check the CPM version will not be set across the entire build. However when/if this change is integrated into all packages then they would all end up using the same CPM script file.

That is true.

> In my view this being able to set the entire build to use a single CPM script feels preferential for developing than to supporting and testing having mixing of CPM versions across packages? It also allows testing if a dependency would have issues upgrading to a new CPM for instance with ease etc.

In the current version, we also don't mix CPM functions. Currently, if a more recent CPM version than used downstream is included, a warning is emitted [notifying the user](https://github.com/cpm-cmake/CPM.cmake/blob/984251f46c058f4573a5f7ce47a6d18a04bf6733/cmake/CPM.cmake#L40) that a newer version is available.

While not essential, we should be aware that this functionality would not persist with the suggested change.